tickets/DM-24503

mareuter opened the pull request.

mareuter committed:
Draft of layout and configuration documentation for deployment.

tribeiro commented on argocd.rst:
```diff
@@ -0,0 +1,82 @@
+ArgoCD Configuration
+====================
+
```
Patrick and I are adding a `note` at the beginning as a placeholder for an abstract. May be a good idea to add something similar here.

tribeiro commented on index.rst:
```diff
@@ -42,20 +42,18 @@
 
```
Patrick and I are adding a note at the beginning as a placeholder for an abstract. May be a good idea to add something similar here.

mareuter replied:
OK. Will do.

tribeiro commented on index.rst:
```diff
@@ -42,20 +42,18 @@
+============
 
-.. note::
+Many of the control system components, but not all, have been developed for deployment via Docker containers. We currently leverage a `docker-compose` mechanism for container deployment to host services. With the proliferation of Kubernetes (k8s) clusters around the Vera C. Rubin Observatory and the associated sites, a new style of deployment and configuration is needed. We will leverage the expe ...
```
Long time ago Tim Jenness suggested to break the lines around 80 characters so, in the future, if you need to change part of a paragraph it would not look like you changed the entire paragraph. I think it would be good if you could do something similar with this document. 

Also, on this paragraph, you mention,

> We currently leverage a `docker-compose` mechanism for container deployment to host services. 

I think this is about to become history pretty soon. I would recommend changing it to something like:

> Initially we leveraged `docker-compose` as a mechanism for prototyping container deployment to host services.

tribeiro commented on helm.rst:
```diff
@@ -0,0 +1,161 @@
+Helm Charts
+===========
+
+The packaging of the CSC deployment starts with a `Helm <https://v2.helm.sh/>`_ chart. We using version 2 of Helm due the current ArgoCD restriction on the Helm version. The code for the charts are kept in the `Helm chart Github repository <https://github.com/lsst-ts/charts>`_. The next two sections will discuss each chart in detail. For a description of the APIs used, consult the `Kubernetes doc ...
```
Is "packaging" the correct word here? 

A couple of typos:

> We using -> We *are* using

When you mention that we are using v2 of Helm I think it would be a good idea to also list the version of argoCD. I would also rephrase that a little bit. A suggestion:

> We are currently adopting version X.X of ArgoCD which works with Helm version 2.

I think you need double `_` for inline links (https://developer.lsst.io/restructuredtext/style.html#external-links). 

API is repeated

> For a description of the APIs used, consult the `Kubernetes documentation <https://kubernetes.io/docs/reference/>`_ ~for the API reference~.

Is this a question? Not sure I understood what you are trying to say here. Can you clarify?

> The Kafka producers may follow a similar pattern, but they are allowed to scale beyond the single pod?

mareuter replied:
The question: The Kafka producers may follow a similar pattern, but they are allowed to scale beyond the single pod? was actually meant for you since you understand the producers better. :-)

mareuter replied:
Hmm. The links work as advertised for me. What behavior did you see?

tribeiro reviewed (changes requested):
Can you please break each paragraph lines in 80 characters? It is harder to review the document with each paragraph being one line long.

tribeiro commented on helm.rst:
```diff
@@ -0,0 +1,161 @@
+   :widths: 15 25
+   :header-rows: 1
+
+   * - YAML Key
```
I didn't like it very much the way the information is displayed in the table. I wonder if it would be better to, instead, add an example yaml file.

mareuter replied:
I can add an example to this, but the table is how most folks document their charts.

tribeiro commented on helm.rst:
```diff
@@ -0,0 +1,161 @@
+volumeMount
+  A CSC that requires access to a physical disk store in order to transfer information into the running container
+
+The chart consists of the Job Kubernetes Workflows API, ConfigMap and PersistentVolumeClaim Kubernetes Config and Storage APIs and VaultSecret `Vault <https://www.vaultproject.io/>`_ API. The Job API is used to provide correct behavior when a CSC is sent of OFFLINE mode, the pod should not restart. The drawback to this is if a CSC dies for an unknown reason, not one caught by FAULT state transiti ...
```
> The Job API is used to provide correct behavior when a CSC is sent of OFFLINE mode, the pod should not restart. The drawback to this is if a CSC dies for an unknown reason, not one caught by FAULT state transition, the pod will not restart and requires startup intervention.

I wonder if this is something we could improve in the future?

mareuter replied:
I found the way to get this to work. Now if the CSC dies unexpectedly, a new pod will start in it's place, but the old one will remain around for debugging. You will still have to bring the component out of STANDBY, but we now get automatic restart in case of unknown failures. I will fix the text.

tribeiro commented on helm.rst:
```diff
@@ -0,0 +1,161 @@
+   :widths: 15 25
+   :header-rows: 1
+
+   * - YAML Key
```
As I mentioned before, I think it would be better to show this in terms of an example yaml file.

mareuter replied:
I will add an example.

mareuter committed:
Reformatting to 80 char lines.

mareuter committed:
Added small abstract.

mareuter committed:
Address review text comments.

mareuter commented:
OK. I've fixed the formatting, added a small abstract and adjusted some of the text comments. I would like to postpone adding the example YAML to the next iteration since I just changed how some of the configuration is handled and don't want to have to do extra work. I promise that there will be examples in the next iteration though. :-) I'd also like to defer fixing the text about the Job API behavior for the next iteration too.

tribeiro reviewed (approved):
looks good. Thanks for the changes.
